File: runtimes/node/nop.js
acc = require('./accessors.js')('http://localhost:6565');

//console.log(acc);
load = process.hrtime();

nop = acc.create_accessor('/webquery/Nop',
		[],
		function(dev) {
			//console.log("----------------------");
			//console.log(dev);
			console.log("ld: " + process.hrtime(load)[1]);

			t1 = process.hrtime();
			dev.StockSymbol(1, function() {
				console.log("q1: " + process.hrtime(t1)[1]);
				t2 = process.hrtime();
				dev.StockSymbol(2, function() {
					console.log("q2: " + process.hrtime(t2)[1]);
					t3 = process.hrtime();
					dev.StockSymbol(3, function() {
						console.log("q3: " + process.hrtime(t3)[1]);
						t4 = process.hrtime();
						dev.StockSymbol(4, function() {
							console.log("q4: " + process.hrtime(t4)[1]);
						});
					});
				});
			});
		},
		function() {
			console.log("err");
		});

File: runtimes/python/nop.py
#!/usr/bin/env python3

import sh

import accessors

import time

t1 = time.time()
print("Start: {} {}".format(t1, 0))

stocktick = accessors.get_accessor_from_server('localhost:6565', '/webquery/Nop')
t2 = time.time()
print("LoadAccessor: {} {}".format(t2, t2-t1))
try:
	stocktick.StockSymbol = 1
	t3 = time.time()
	print("Query1: {} {}".format(t3, t3-t2))

	stocktick.StockSymbol = 2
	t4 = time.time()
	print("Query2: {} {}".format(t4, t4-t3))

	stocktick.StockSymbol = 3
	t5 = time.time()
	print("Query3: {} {}".format(t5, t5-t4))

	stocktick.StockSymbol = 4
	t6 = time.time()
	print("Query4: {} {}".format(t6, t6-t5))
finally:
	# Hack until I understand bond better
	sh.killall('node')
